### PR TITLE
WIP : add support for tasks

### PR DESCRIPTION
**Pull request by kdw2060**
## Description

Hi, I was missing support for tasks, so tried to add support.
Be aware: I'm new to Python and also couldn't test as I haven't figured out where the pygrocy files are stored within the Hass.io filesystem. Do you know where these go?

I basically copied the Chore-related classes etc and made Tasks-related ones.
For naming of properties and actions I stuck to the Grocy api naming, just like you did for all the other objects.

Without having been able to test, there's probably still errors in this code. I would be happy to correct those, or off course feel free to do so yourself.

I did the same for the Grocy component, will start up pull request there too.

## Checklist
  - [ ] The code change is tested and works locally.
  - [ ] tests pass. **Your PR won't be merged unless tests pass**
  - [ ] There is no commented out code in this PR

### CODE REVIEW

**Comment by BlueBlueBlob** [comment]
Hi ! 
Thx for your work. Can you correct build issues ?
If you need help, ask :) 👍

**Comment by kdw2060** [comment]
Hi,

i don't fully understand the id-related errors from CodeFactor. Should I add a '_' before 'id'?
For the chores the api response object has 'chore_id', but for the tasks it's just 'id'.

**Comment by BlueBlueBlob** [comment]
> Hi,
> 
> i don't fully understand the id-related errors from CodeFactor. Should I add a '_' before 'id'?
> For the chores the api response object has 'chore_id', but for the tasks it's just 'id'.

Because id is a keyword like int for if etc... Change with task_id

**Comment by BlueBlueBlob** [comment]
To complete my comment : you get id with parse_json.get('id'), between '' you can use id but not as a variable.

**Comment by kdw2060** [comment]
Hi, made the change. Figured out the TaskResponse and Task objects would have to be changed accordingly too, so changed the variable name there as well.

I also corrected the errors that resulted in the duplicate code error messages, but strangely enough these still appear.

**Comment by BlueBlueBlob** [comment]
Try to change property order in grocy OR grocy_api.
(Not sure 😹)

**Comment by BlueBlueBlob** [comment]
Good job 👍

**Comment by BlueBlueBlob** [comment]
Now you have to add testing for new features.
Check /test/
Ask if you need help

**Comment by BlueBlueBlob** [comment]
You need to install https://github.com/SebRut/pygrocy/blob/master/requirements-dev.txt
```
pip install -r requirements-dev.txt
```
When you think tests are done and before git push, you can check result :
```
coverage run --source=pygrocy setup.py test
coverage report
```

**Comment by kdw2060** [comment]
Did that, test fails. This is a bit out of my league I'm afraid. I have no experience with build tools and automated tests. I'm just an amateur web-dev, accustomed to working with smallish javascript projects.
_(contemplated just making a custom lovelace card in js that does all this, but tried to contribute here instead so it would be more useful to more people)_

**Comment by BlueBlueBlob** [comment]
> Did that, test fails. This is a bit out of my league I'm afraid. I have no experience with build tools and automated tests. I'm just an amateur web-dev, accustomed to working on smallish javascript projects.
> _(contemplated just making a custom lovelace card in js that does all this, but tried to contribute here instead so it would be more useful to more people)_

Don't worry and don't be afraid. I was in shoes ;)
Try to read a simple function in test_grocy
It means 3 test functions in most of case : test_yournewfunction_valid, test_yournewfunction_invalid_no_data and test_yournewfunction_invalid_missing_data

I'm looking for a good exemple for one of your add. 
I know it take some time, but it's very useful. And when you know how it works, it's most of copy/paste

👍

**Comment by BlueBlueBlob** [comment]
Ok
Check shopping_list implementation in test_grocy
You have to simulate what you send and what you receive.

No so hard, just time :/

File: test/test_tasksResponse.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+class TestTasksResponse(TestCase):
+    def test_parse(self):
+        input_json = """{
+            "id": 0,
+            "category_id": 0,
+            "name": "string",
+            "description": "string",
+            "due_date": "2019-05-04T11:31:04.563Z",
+            "done": 0,
+            "done_timestamp": ,
```
Comment (SebRut) [inline test/test_tasksResponse.py]: This line could just be ommitted.

File: test/test_tasksResponse.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            "assigned_to_user_id": 0
+        }"""
+
+        response = TasksResponse(json.loads(input_json))
+
+        assert response.task_id == 0
```
Comment (SebRut) [inline test/test_tasksResponse.py]: could you add assertions for the remaining fields as far as possible?

**Comment by kdw2060** [comment]
Hi I'm closing this request. I didn't have the time to dive deeper into python and set up a complete Home Assistant development environment, so I just resorted to writing a custom card in javascript for now.

**Pull request closed without merging**